### ISSUE TITLE
Consider hosting on Netlify

### ISSUE THREAD

**Issue by HugoGiraudel** [HugoGiraudel/sass-guidelines#313]
[Netlify](https://www.netlify.com) is a hosting solution specialised in static sites.

## How does it work?

You basically associate Netlify to a GitHub repository, and whenever a push is made on a specific branch, Netlify runs whatever command it is asked to run, and redeploy whatever folder it is asked to deploy.

For instance:
* Repository: https://github.com/hugogiraudel/sass-guidelines
* Branch: `gh-pages`
* Command: `jekyll build`
* Folder: `_site`

## What for?

- We would not be entitled to run Jekyll in safe mode anymore, which would leave room for Jekyll plugins usage, which would be better than hacking around Liquid to come to our ends.
- We could enable SSL and serve the site over HTTPS, which not only looks and feels better but also would allow us to use Service Workers for offline support and the like.
- We could run any kind of tool we want pre/post-build, which could be useful for syncing translations, optimising CSS with criticalCSS/unCSS, and anything else.

## Where is the catch?

But, it would cost $7.5 per month, or $90 a year. Which is okay, I’d say.

---

Any thoughts?

**Comment by HugoGiraudel** [comment]
Alright, so I have been considering Netlify for the past few hours and I went so far to test the free plan just for the sake of it. I created [a new GitHub repository](https://github.com/HugoGiraudel/netlify-jekyll) with a very basic Jekyll setup. I could [deploy it live](http://cobbler-references-27543.netlify.com/) in a matter of —literally— seconds.

<img width="1194" alt="screen shot 2016-02-06 at 22 13 34" src="https://cloud.githubusercontent.com/assets/1889710/12869409/f659d43a-cd1e-11e5-8863-b4a894a3ea03.png">

Next, I wanted to see if I could run some post-build script on their servers. I created a [shell script](https://github.com/HugoGiraudel/netlify-jekyll/blob/master/bin/build) that runs unCSS, criticalCSS, cleanCSS, uglifyJS and everything. I added [a npm task](https://github.com/HugoGiraudel/netlify-jekyll/blob/master/package.json#L3) to run this script and set up Netlify to run this task before publishing.

<img width="676" alt="screen shot 2016-02-06 at 22 13 52" src="https://cloud.githubusercontent.com/assets/1889710/12869411/ff0a98c6-cd1e-11e5-97cd-e667cc939a48.png">

The result is great: everything gets executed seamlessly and it is makes criticalCSS and inlined JS utilities very easy. I can see that Netlify also hosts assets on their CDN which is pretty rad.

<img width="1108" alt="screen shot 2016-02-06 at 22 14 52" src="https://cloud.githubusercontent.com/assets/1889710/12869416/333fd55c-cd1f-11e5-9ecb-d170896e2c1b.png">

Long story short, I think it would be pretty nice to move Sass Guidelines to Netlify, as mentioned in the initial post. This would solve all the current problems we have with Jekyll for a minor cost that I am willing to pay.

**Comment by HugoGiraudel** [comment]
## Roadmap

1. Clone the sources (currently on `gh-pages`) into `master`
1. Create a new site on Netlify linked to `master`
1. Password-protect the new site while we make sure everything works correctly
1. Add build script and set up Netlify to use it
1. Remove password from new site
1. Set up DNS redirect from OVH to Netlify
1. Delete the gh-pages branch (once DNS redirect is effective )
1. Ask authors of all existing PRs to renew their PR against the `master` branch

**Comment by HugoGiraudel** [comment]
Moved to #362.
